### PR TITLE
Include gdpr TripleLift

### PR DESCRIPTION
**Pull request by brittanyzellman**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
Upgraded to 1.12 to include SRA capability and GDPR compliance module.

- contact email of the adapter’s maintainer
bzellman@triplelift.com
- [x] official adapter submission

### CODE REVIEW

**Comment by mkendall07** [comment]
This pull request **introduces 1 alert** when merging 45efe0ee374a8d4a55aa4d565328acc3273313a6 into 476cad8d9287beb2c5054d9b164a05717e4437a5 - [view on lgtm.com](https://lgtm.com/projects/g/prebid/Prebid.js/rev/pr-305b0526dae9dd3044c37572e887b290cfbdcfeb)

**new alerts:**

* 1 for Useless conditional

---

*Comment posted by [lgtm.com](https://lgtm.com)*

**Comment by mkendall07** [comment]
This pull request **introduces 1 alert** when merging 2ebff0cd73b9e515800a67daa89d1b1bc8bf6551 into 476cad8d9287beb2c5054d9b164a05717e4437a5 - [view on lgtm.com](https://lgtm.com/projects/g/prebid/Prebid.js/rev/pr-e777d9809e53c9d9293e8b4f31d4cd7808f48060)

**new alerts:**

* 1 for Useless conditional

---

*Comment posted by [lgtm.com](https://lgtm.com)*

**Comment by brittanyzellman** [comment]
@mkendall07 I would like to reach out various publishers with a timeline of when they can expect the adapter to be deployed. Pending all tests/approval, when can I expect a merge and deploy?
Thanks!

File: integrationExamples/gpt/custom.html [hunk 1]
```diff
@@ -0,0 +1,186 @@
+<html>
```
Comment (mkendall07) [inline integrationExamples/gpt/custom.html]: please remove this file. We already have a gdpr hello world example.

**Comment by YOzaz** [comment]
We're a publisher, and waiting for TripleLift to be merged... hope it'll be today?

**Comment by ehoch** [comment]
@brittanyzellman This is causing

``Error processing command : Cannot read property 'gdprApplies' of undefined TypeError: Cannot read property 'gdprApplies' of undefined```

If the consent module is included, but not configured (like we have in the case of our non-EU traffic).  Can you add a check for existence of bidderRequest.gdprConsent ?

**Comment by ptim** [comment]
@brittanyzellman not sure, but merge may be waiting on removal of integrationExamples/gpt/custom.html, per:

> please remove this file. We already have a gdpr hello world example.

https://github.com/prebid/Prebid.js/pull/2663#discussion_r192192453

**Comment by brittanyzellman** [comment]
All irrelevant files are deleted and adapter passes all checks. Still waiting for a timeline as to when publishers can expect this to be merged and deployed? @mkendall07

**Comment by ptim** [comment]
I'm a pub using triplelift - can I request some documentation for configuring native ad units under Prebid v1?  

It appears that it's necessary to send bid requests `mediaType.banner` with a [1,1] size rather than using the native ad configuration... I only discovered this via experimentation - can't see it documented anywhere (there's no readme for this adapter, and no entry in the prebid docs repo)

TIA

**Comment by brittanyzellman** [comment]
@ptim please reach out to your point of contact at Triplelift. They will have all relevant documentation on how to integrate. I'm assuming that you are already integrated with us through Prebid?

**Comment by YOzaz** [comment]
@ptim Triplelift does not support "real" `pbjs` native bidding, but rather is a wrapper for `[1,1]` auto-expanding ad. Same is for Sharethrough, for example.

File: modules/tripleliftBidAdapter.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    },
+    bids: [
+    {
+        bidder: 'triplelift',
+        params: {
+           inventoryCode: 'forbes_main',
+           floor: 1.009
+        }
+    },{
+        bidder: 'appnexus',
```
Comment (mkendall07) [inline modules/tripleliftBidAdapter.md]: any reason you included an AppNexus bid in your test?

File: modules/tripleliftBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
+import { BANNER } from 'src/mediaTypes';
+import { registerBidder } from 'src/adapters/bidderFactory';
+import * as utils from 'src/utils';
+
+const BIDDER_CODE = 'triplelift';
+const STR_ENDPOINT = document.location.protocol + '//tlx.3lift.com/header/auction?';
+var applies = true;
```
Comment (mkendall07) [inline modules/tripleliftBidAdapter.js]: prefer `let` over `var`

Comment (mkendall07) [reply modules/tripleliftBidAdapter.js]: a better name here would be `gdprApplies`

File: modules/tripleliftBidAdapter.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+var consentString = null;
+
+export const tripleliftAdapterSpec = {
+
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+  aliases: ['triplelift'],
```
Comment (mkendall07) [inline modules/tripleliftBidAdapter.js]: not necessary to alias your own code.

**Comment by mkendall07** [comment]
@brittanyzellman 
See comments above.

File: modules/tripleliftBidAdapter.md (hunk 2 above)
Comment (brittanyzellman) [reply modules/tripleliftBidAdapter.md]: wanted to see in prebid debug that I was seeing multiple bidders, would you like me to remove?

**Comment by brittanyzellman** [comment]
@mkendall07 all changes were made, I had AN in the sample for testing just so to see whether other bidders were able to load and bid but I've removed it from the md file. 
Thanks!

**Comment by mkendall07** [comment]
@brittanyzellman 
Thanks for the changes. One last thing, please open a PR to add 
```
prebid_1_0_supported : true
gdpr_supported: true
```
To the header of this file:
https://github.com/prebid/prebid.github.io/blob/master/dev-docs/bidders/triplelift.md

**Comment by mkendall07** [comment]
@brittanyzellman 
also please see: https://github.com/prebid/Prebid.js/pull/2833

**Pull request merged**